Alter use of the --verbose flag for `fuelup check`

JoshuaBatty opened the pull request:
The output of `fuelup check --verbose` is what I would expect `fuelup check` to output. Before this change, `fuelup check` outputted a fraction of the installed components on the system. ~~I feel like this just leads to confusion amongst users so it's been removed.~~

I added the -v flag back in. The default behaviour is to only show fuelup or components need updating. The -v flag shows all components and prints that they are up to date. See screen shot below. 

The rest of the diff is just cleaning up imports and doing clippy. 

![Screenshot 2024-06-25 at 1 59 44 pm](https://github.com/FuelLabs/fuelup/assets/1289413/f7cbe5ea-1433-425e-8ac0-3a2bbbe7b664)

closes #553

sdankel commented:
I agree that we shouldn't need a verbose flag for `fuelup check`, but I also think we should condense its output. I think it should show 1 line per installed toolchain, rather than showing every plugin in the toolchain, similar to rustup.

![image](https://github.com/FuelLabs/fuelup/assets/47993817/d628a8ac-42af-4cac-9f65-3518a8b9f004)

![image](https://github.com/FuelLabs/fuelup/assets/47993817/236f7f92-8fde-432d-a25f-87db1ecd359c)

Currently it prints a download bar even when everything is up to date. On that note, `check` should never download anything, it should only tell the user if something can be updated, so that's a bug.

JoshuaBatty commented:
Thanks Sophie. I like the idea of mimicking what `cargo check` does here. I'll also look into avoiding need to download anything for this check. Putting into draft mode until this is finished.

JoshuaBatty commented:
Thanks @sdankel this should be ready now. I added the -v flag back in so it prints if the component of fuelup is up to date, but the default behaviour is now similar to rustup and only shows if something is out of date. 

I also removed the download bar. We still need to download the toml file from git to check the versions but it's only a handful of kb so showing a download bar seems unnecessary.

sdankel commented on src/ops/fuelup_check.rs:
```diff
@@ -53,95 +49,94 @@ fn format_version_comparison(current_version: &Version, latest_version: &Version
     let latest_package_versions = collect_package_versions(dist_channel);
-
     let toolchain = Toolchain::new(toolchain)?;
+    info!("{}: {}", bold("Toolchain: "), &toolchain.name);
```
```suggestion
    info!("{}{}", bold("Toolchain: "), &toolchain.name);
```

![image](https://github.com/FuelLabs/fuelup/assets/47993817/6d9fe984-0da4-42e7-a016-f6421a04798c)

I get the need for this label, because with all of the components listed below it it's hard to see that that's the toolchain name. It's a lot more readable without the download bar! 👍 

Since we have the `-v` added back, I wonder if we should go further with condensing the output in non-verbose mode. It could be like this:

```
latest-aarch64-apple-darwin - Up to date
nightly-aarch64-apple-darwin - Update available (12 updates)
```

JoshuaBatty replied:
I like this idea :)

JoshuaBatty replied:
putting into draft mode until I complete this

JoshuaBatty replied:
done in [05a3363](https://github.com/FuelLabs/fuelup/pull/631/commits/05a336343b91089d0b0156d0567c99053c284121)

sdankel reviewed (commented):
Looking much better! One thing I noticed is when I have 1 toolchain that needs updates and 1 that is up to date, the formatting for the up-to-date one is off, and doesn't say up-to-date:

![image](https://github.com/FuelLabs/fuelup/assets/47993817/06fc6006-f2d5-42b6-ae05-f812222a661a)

JoshuaBatty commented:
Thanks @sdankel i've update the description with a screenshot of what the default and verbose options produce.

fuel-service-user commented:
### [LCOV](https://github.com/marketplace/actions/report-lcov) of commit [<code>9bde78c</code>](631/commits/9bde78c5bb31b65ecfa46c0d6b7cd1dd44303cf6) during [CI #2025](../actions/runs/9669662618)
<pre>Summary coverage rate:
  lines......: 81.9% (2315 of 2827 lines)
  functions..: 44.4% (363 of 818 functions)
  branches...: 58.6% (278 of 474 branches)

Files changed coverage rate: n/a</pre>

sdankel reviewed (approved):
LGTM 🚀
